File: LEET-CODE/1689.Partitioning_Into_Minimum_Number_Of_Deci-Binary_Numbers/Partitioning_Into_Minimum_Number_Of_Deci-Binary_Numbers.cpp
#include <bits/stdc++.h>
using namespace std;

// Define a class Solution
class Solution {
public:
    // Define a method named minPartitions that takes a string n as input and returns an integer.
    int minPartitions(string n) {
        // Initialize a character variable c to the first character of the input string.
        char c=n[0];
        // Iterate over all the characters in the input string.
        for(char i : n){
            // Update the value of c to the maximum character encountered so far.
            if(c<i){
            c=i;
            }
        }
        // Return the integer equivalent of the character c.
        return c-'0';
    }
};

File: LEET-CODE/1689.PARTITIONING_INTO_MINIMUM_NUMBER_OF_DECI_BINARY_NUMBERS/Partitioning_Into_Minimum_Number_Of_Deci-Binary_Numbers.py
# Define a class named Solution.
class Solution:
    
    # Define a method named minPartitions that takes a string n as input and returns an integer.
    def minPartitions(self, n: str) -> int:
        # Convert the input string into a list of characters, then find the maximum character and convert it into an integer.
        return int(max(list(n)))
